crates/sui-core/src/consensus_adapter.rs

```diff
@@ -212,17 +256,37 @@ impl ConsensusAdapter {
         // refreshed frequently enough to make sure this is Byzantine-resistant
     }
 
+    /// This method is called externally to begin reconfiguration
+    /// It transition reconfig state to reject new certificates from user
+    /// ConsensusAdapter will send EndOfPublish message once pending certificate queue is drained
+    pub fn begin_reconfiguration(self: &Arc<Self>) -> SuiResult {
```

Comment:
I think a better name is "close_epoch”, because from this on the next step is actually constructing the last checkpoint

Reply:
Renamed